typescript/pragma-scripts/src/dispatcher/add_feeds.ts

```diff
@@ -0,0 +1,80 @@
+import { Command, type OptionValues } from "commander";
+import type { Contract } from "starknet";
+import {
+  buildAccount,
+  Deployer,
+  ensureSuccess,
+  getDeployedAddress,
+  STARKNET_CHAINS,
+} from "pragma-utils";
+
+function parseCommandLineArguments(): OptionValues {
+  const program = new Command();
+  program
+    .name("add-feeds")
+    .description("CLI to update multiple Pragma data feeds")
+    .requiredOption(
+      "--feed-ids <feed_ids...>",
+      "IDs of the data feeds to update",
+    )
+    .requiredOption(
+      "--chain <chain_name>",
+      "Name of the target chain (e.g., pragmaDevnet)",
+    )
+    .parse(process.argv);
+
+  const options = program.opts();
+  if (!STARKNET_CHAINS.includes(options.chain)) {
+    console.error(
+      `Error: Invalid starknet chain, must be in ${STARKNET_CHAINS.toString()}.`,
+    );
+    process.exit(1);
+  }
+  return options;
+}
+
+async function addFeed(
+  pragmaDispatcher: Contract,
+  account: Deployer,
+  feedId: string,
+) {
+  try {
+    console.log(`\t ⏳ Adding feed ${feedId}...`);
+    const invoke = await pragmaDispatcher.invoke("add_feed", [feedId]);
+    await account.waitForTransaction(invoke.transaction_hash);
+
+    const receipt = await account.getTransactionReceipt(
+      invoke.transaction_hash,
+    );
+    await ensureSuccess(receipt, account.provider);
```

Comment:
this code block could be extracted and replaced everywhere its used, also make sure to repeatly call getTransactionReceipt as receipt might not directly be available

Reply:
it is called after `waitForTransaction` so should be gud?

Reply:
yea should be indeed